### ISSUE TITLE
Incorrect behaviour with innerHTML and <textarea>

### ISSUE THREAD

**Issue by nielsdos** [lexbor/lexbor#238]
I adapted the code from the innerHTML example:

```c
/*
 * Copyright (C) 2018 Alexander Borisov
 *
 * Author: Alexander Borisov <borisov@lexbor.com>
 */

#include "base.h"


int
main(int argc, const char *argv[])
{
    lxb_html_element_t *element;
    lxb_html_body_element_t *body;
    lxb_html_document_t *document;

    static const lxb_char_t html[] = "<textarea>contents</textarea>";
    size_t html_len = sizeof(html) - 1;

    static const lxb_char_t inner[] = "</textarea>foo";
    size_t inner_len = sizeof(inner) - 1;

    /* Parse */
    document = parse(html, html_len);

    /* Print Incoming Data */
    PRINT("HTML:");
    PRINT("%s", (const char *) html);
    PRINT("\nTree after parse:");
    serialize(lxb_dom_interface_node(document));

    /* Get BODY element */
    body = lxb_html_document_body_element(document);

    PRINT("\nHTML for innerHTML:");
    PRINT("%s", (const char *) inner);

    element = lxb_html_element_inner_html_set((lxb_html_element_t *) lxb_dom_node_first_child(&lxb_html_interface_element(body)->element.node),
                                              inner, inner_len);
    if (element == NULL) {
        FAILED("Failed to parse innerHTML");
    }

    /* Print Result */
    PRINT("\nTree after innerHTML set:");
    serialize(lxb_dom_interface_node(document));

    /* Destroy all */
    lxb_html_document_destroy(document);

    return 0;
}
```

This code parses a textarea element, and then changes its innerHTML to `</textarea>foo`.
I expect the innerHTML to then equal to a  single text node `</textarea>foo`, but it is equal to a single text node `foo` instead.

The equivalent Javascript (in both Firefox and Chrome) do result in the right result:
```js
dom=(new DOMParser).parseFromString("<textarea>foo</textarea>","text/html")
dom.body.firstElementChild.innerHTML="</textarea>text"
console.log(dom.body.firstElementChild.innerHTML) // </textarea>text
console.log((new XMLSerializer).serializeToString(dom.body.firstElementChild)) // <textarea xmlns="http://www.w3.org/1999/xhtml">&lt;/textarea&gt;text</textarea>
```

**Comment by lexborisov** [comment]
Hi @nielsdos 

You found the bug! Fixed in [f0539d1](https://github.com/lexbor/lexbor/commit/f0539d137a9760d724dd378fad1c9fba1b083419).

Thanks for the report!

**Comment by lexborisov** [comment]
Thank you for the reports, it allows the library to become a better library.
